.github/workflows/ci.yml

```diff
@@ -102,6 +112,16 @@ jobs:
         SqlLocalDB info MSSQLLocalDB
         sqlcmd -S "(localdb)\MSSQLLocalDB" -Q "create database cakephp;"
 
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+    - name: Cache composer dependencies
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
```

Comment:
Our composer.json rarely changes. Wouldn't this prevent us from having tests run against new patch releases of our dependencies as they are released?

Reply:
Yeah ideally the `composer.lock` file should be used, but we don't have that committed in the repo.

Reply:
> https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

Unused cache keys are removed after 7 days but I don't see any info about forcibly clearing cache. So we can just take a break of 7 days every month or so to clear the cache :smile:.

Reply:
Won't our key always be 'used'? Could we include a date based component to the cache key so it automatically rotates once a week?